src/utils/flow.js

```diff
@@ -46,6 +46,21 @@ export function convertFlowObjectTypeAnnotation(
       };
     }
 
+    if (value.type === "ArrayTypeAnnotation" && value.elementType) {
+      const children = value.elementType.id && flowTypes[value.elementType.id.name]
+        ? flowTypes[value.elementType.id.name]
+        : value.elementType;
+
+      return {
+        ...obj,
+        [key]: {
+          type: "array",
+          nullable: property.optional,
+          children: children.properties ? convertFlowObjectTypeAnnotation(children, flowTypes) : null
```

Comment:
What I need is something that can basically do that whole block without it pushing it into keys, since the `GenericTypeAnnotation` is also valid for that. That's why I chose duck typing. I'm happy to expand on that and make it something reusable across the board.